shell/platform/darwin/common/framework/Source/FlutterNSBundleUtils.mm

```diff
@@ -26,11 +26,38 @@
   return nil;
 }
 
-NSBundle* FLTFrameworkBundleWithIdentifier(NSString* bundleID) {
-  NSBundle* bundle = FLTFrameworkBundleInternal(bundleID, NSBundle.mainBundle.privateFrameworksURL);
-  if (bundle != nil) {
-    return bundle;
+NSBundle* FLTGetApplicationBundle() {
+  NSBundle* mainBundle = [NSBundle mainBundle];
+  // App extension bundle is in Runner.app/PlugIns/Extension.appex.
+  if ([mainBundle.bundleURL.pathExtension isEqualToString:@"appex"]) {
```

Comment:
Can we unit test these utils, especially things like this alternate codepath?

Reply:
Oh no, I forgot to push the unittests from my home computer. I will push the commit when I get home.